### DISCUSSION TITLE
Proposal: return different Path for SpecialFolder.Fonts on Linux

### DISCUSSION THREAD

**Discussion by Miepee** [dotnet/runtime#63217] [General]
Currently on Linux, `~/.fonts` is returned for SpecialFolder.Fonts
However, [that directory](https://wiki.archlinux.org/title/Font_configuration#Font_paths) is [deprecated](https://cgit.freedesktop.org/fontconfig/commit/?id=8c255fb), and `$XDG_DATA_HOME/fonts` should be used instead for local user fonts.
What are the thoughts on this, should I do a PR for that?

**Comment by huoyaoyuan** [comment]
> that directory is deprecated

Is it widely adopted? 

I don't expect `SpecialFolder` to get evolved. It was originally for Windows, and on non Windows platforms it has to return "somewhere" to make libraries not crashing. The meanings of the folders won't be the same on all platforms.

**Comment by Miepee** [comment]
> Is it widely adopted?

From what I've seen yes, but I'm also just a sole Arch user. The ArchWiki says that `~/.fonts` will be stopped read entirely on future releases, Gentoo wiki says that it's not being read at all anymore, the Ubuntu Wiki on Fonts is very out of date and Debian Wiki seems as well as that folder isn't even mentioned to be legacy folder. I've seen some programs create a custom fontconfig config file to include `~/.fonts` back so that fontconfig reads them.
From what it's worth,`$XDG_DATA_HOME/fonts` is the recommended path by fontconfig.

> The meanings of the folders won't be the same on all platforms.

Could documentation be adjusted for that then? I can't see anything [here](https://docs.microsoft.com/en-us/dotnet/api/system.environment.specialfolder?view=net-6.0) that suggests that meanings would be different on other OS.
